test/unit/alignment/exception_test.cpp

```diff
@@ -0,0 +1,37 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+#include <gtest/gtest.h>
+
+#include <seqan3/alignment/exception.hpp>
+
+TEST(alignment_exception, invalid_alignment_configuration)
+{
+
+    auto throw_lambda = []() {
+        throw seqan3::invalid_alignment_configuration{"test"};
+    };
+
+    EXPECT_THROW(throw_lambda(), seqan3::invalid_alignment_configuration);
+
+    std::string message;
+
+    try
+    {
+        throw_lambda();
+    }
+    catch (seqan3::invalid_alignment_configuration & ex)
+    {
+        message = ex.what();
+    }
+    catch (...)
+    {
+        FAIL();
+    }
```

Comment:
what are you testing here? :thinking:

Reply:
I am testing that the message is correctly passed to the exception object. And in the case of something completely unexpected it should fail.

Reply:
You mean you wrote a 25 line testing for the one-line constructor of your exception that only forwards to a std::exception? Isn't that a little too much?
Couldn't you rather test a case where the exception is thrown by your algorithm?

Reply:
One test per class.